### DISCUSSION TITLE
few ideia

### DISCUSSION THREAD

**Discussion by josef2600** [pschatzmann/arduino-audio-tools#1035] [Q&A] [answered]
not to bug you, but i thought of few thoughts! if i may:
The readme is too large. It has to be optimized by cutting it to separate parts.
The first part should be explaining about hardware’s and how they work and their settings.
Before going to esp32 hardware, you have to explain about connection hardware. For Serial.begin(115200); it is too slow. It has to be at least 512000 baud rate. Low speed can cause the micro to crash, usually because it overflows its memory or conflicting with other instructions at the same time. But not all USB to serials can handle speeds over 256000, so it is extremely important to say it and explain it. 
So far:
the chips that work: (high speed – work with 921600)
FT232 - MCP2200 (best for electronic – maybe I am a bit biased about microchip too!) - FT312D-32L1C-T (android support-complex) - FT4232HL (very complex - very fast – Bridge and USB to UART) - 
the chips that do not work: (low speed – max 256000)
CH340 (very bad and noisy) - 
These are just some chipsets I have worked with and are famous. They have many variants and different configurations. But most important thing is their datasheet. User has to read it. And remember China has mostly fake chipset and may not work correctly. 

thank you.
Best regards.

**Comment by pschatzmann** [comment, marked as answer]
It is the intention to have the Readme this long to scare away the people that don't know what they are doing. After all this is a general purpose Arduino audio library for multiple processor architectures and not a  finished audio player!

The ESP32 is just my main development platform that provides most of the functionality.

Did you read the Wiki ?
